server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aeternity/AeternityWalletGenerator.java

```diff
@@ -0,0 +1,93 @@
+/*************************************************************************************
+ * Copyright (C) 2014-2019 GENERAL BYTES s.r.o. All rights reserved.
+ *
+ * This software may be distributed and modified under the terms of the GNU
+ * General Public License version 2 (GPL2) as published by the Free Software
+ * Foundation and appearing in the file GPL2.TXT included in the packaging of
+ * this file. Please note that GPL2 Section 2[b] requires that all works based
+ * on this software must also be made publicly available under the terms of
+ * the GPL2 ("Copyleft").
+ *
+ * Contact information
+ * -------------------
+ *
+ * GENERAL BYTES s.r.o.
+ * Web      :  http://www.generalbytes.com
+ *
+ ************************************************************************************/
+package com.generalbytes.batm.server.extensions.extra.aeternity;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.generalbytes.batm.server.extensions.IExtensionContext;
+import com.generalbytes.batm.server.extensions.IPaperWallet;
+import com.generalbytes.batm.server.extensions.IPaperWalletGenerator;
+import com.kryptokrauts.aeternity.sdk.domain.secret.impl.BaseKeyPair;
+import com.kryptokrauts.aeternity.sdk.domain.secret.impl.MnemonicKeyPair;
+import com.kryptokrauts.aeternity.sdk.exception.AException;
+import com.kryptokrauts.aeternity.sdk.service.keypair.KeyPairService;
+import com.kryptokrauts.aeternity.sdk.service.keypair.KeyPairServiceFactory;
+import com.kryptokrauts.aeternity.sdk.util.EncodingUtils;
+
+public class AeternityWalletGenerator implements IPaperWalletGenerator {
+
+    private static final Logger log = LoggerFactory.getLogger("batm.master.BitcoinWalletGenerator");
+    //private String prefix;
+    private IExtensionContext ctx;
+
+    public AeternityWalletGenerator(String prefix, IExtensionContext ctx) {
+        //this.prefix = prefix;
+        this.ctx = ctx;
+    }
+
+    @Override
+    public IPaperWallet generateWallet(String cryptoCurrency, String oneTimePassword, String userLanguage) {
+    	final KeyPairService keyPairService = new KeyPairServiceFactory().getService();
+    	MnemonicKeyPair master = null;
+    	BaseKeyPair credentials = null;
+		try {
+			master = keyPairService.generateMasterMnemonicKeyPair(oneTimePassword);
```

Comment:
This is very weak one time password has only few letters. Please make the key generation stronger. One time password should be used only as a ZIP password. Preferably use SecureRandom.

Reply:
ok will generate new stronger password